### PR TITLE
Rooms work now, sort of

### PR DESCRIPTION
**Pull request by Sven-dev**
Player gets softlocked when moving into another room now, because the doors close behind them and the room cant be cleared because there are no enemies yet.

### CODE REVIEW

File: RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs [hunk 1]
```diff
@@ -1,39 +1,112 @@
...
+using UnityEngine;
+using System.Collections;
+
+namespace nl.SWEG.RPGWizardry.GameWorld
+{
+    [RequireComponent(typeof(Camera))]
+    public class CameraManager : SingletonBehaviour<CameraManager>
+    {
+        #region Fields
+        private ScreenFade screenFader;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs:12]: Summary

File: RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs [hunk 2]
```diff
@@ -1,39 +1,112 @@
...
+        [Range(0.00f, 2f)]
+        private float FadeTime;
+        #endregion
+
+        #region Methods
+        #region Public
+        /// <summary>
+        /// Fade the camera visibility between 2 values.
+        /// </summary>
+        /// <param name="from">From.</param>
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs:37]: explain what these values mean

File: RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs [hunk 3]
```diff
@@ -1,39 +1,112 @@
...
+            screenFader.enabled = true;
+            LeanTween.value(gameObject, UpdateShader, from, to, FadeTime);
+        }
+        #endregion
+
+        #region Private
+        /// <summary>
+        /// Updates the shader used for room transitioning.
+        /// </summary>
+        /// <param name="value">A value between 0 and 1.</param>
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs:63]: Explain what this value means wrt visuals

File: RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs [hunk 4]
```diff
@@ -1,39 +1,112 @@
...
+        private void Update()
+        {
+
+            Camera.enabled = true;
+            if (!PlayerManager.Exists)
+                return;
+
+            Transform playerTF = PlayerManager.Instance.transform;
+            transform.position = new Vector3(
+                Mathf.Round(playerTF.position.x * 1000.0f) / 1000.0f,
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/CameraManager.cs:117]: Math.Round(value, decimals) might be more appropriate

File: RPG Wizardry/Assets/Scripts/GameWorld/FloorManager.cs [hunk 5]
```diff
@@ -1,71 +1,155 @@
...
+                PlayerManager.Instance.transform.position = destination.transform.position;
+
+
+            //Enable the new room
+            destination.Room.Enable();
+            activeRoom = destination.Room;
+
+            //Fade the screen back in
+            CameraManager.instance.Fade(0, 1);
+            while (CameraManager.instance.Fading)
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/FloorManager.cs:98]: yield return new WaitUntil(!Fading);

File: RPG Wizardry/Assets/Scripts/GameWorld/FloorManager.cs [hunk 6]
```diff
@@ -1,71 +1,155 @@
...
+            //Make sure the player can move again
+            GameManager.Instance.Locked = false;
+
+            //Activate enemies in new room
+            if (!activeRoom.Cleared)
+            {
+                //Spawn enemies
+
+                //close doors
+                yield return new WaitForSeconds(0.25f);
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/FloorManager.cs:112]: 'magic' number used in multiple places. Create a const value.

File: RPG Wizardry/Assets/Scripts/GameWorld/GameLogic/GameManager.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using nl.SWEG.RPGWizardry.Utils.Behaviours;
+
+namespace nl.SWEG.RPGWizardry
+{
+    public class GameManager : SingletonBehaviour<GameManager>
+    {
+        #region Variables
+        public GameState GameState;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/GameLogic/GameManager.cs:11]: private set
summaries

File: RPG Wizardry/Assets/Scripts/GameWorld/Room.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace nl.SWEG.RPGWizardry.GameWorld
+{
+    public class Room : MonoBehaviour
+    {
+        public bool Cleared;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/Room.cs:10]: private set ( or get { check enemy count } )

File: RPG Wizardry/Assets/Scripts/GameWorld/ScreenFade.cs [hunk 9]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using nl.SWEG.RPGWizardry.Player;
+using nl.SWEG.RPGWizardry.Utils.Functions;
+
+namespace nl.SWEG.RPGWizardry.GameWorld
+{
+    [RequireComponent(typeof(Camera))]
+    public class ScreenFade : MonoBehaviour
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/GameWorld/ScreenFade.cs:10]: Clean

File: RPG Wizardry/Assets/Scripts/Player/Combat/AimingManager.cs [hunk 10]
```diff
@@ -50,28 +50,32 @@ private void Update()
...
+                //Get location to look at
+                Vector3 lookPos = PlayerManager.Instance.InputManager.State.AimingData;
+                float angle = Mathf.Atan2(lookPos.y, lookPos.x) * Mathf.Rad2Deg;
+                //Rotate to look at mouse/controller direction
+                BookPivot.rotation = Quaternion.AngleAxis(angle, Vector3.forward);
+
+                //Change rotation parameter in animator, changing sprites
+                bookAnimator.SetFloat("Rotation", BookPivot.rotation.z);
+
+                //Change Z position, rotating around player appropriately
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Player/Combat/AimingManager.cs:65]: Is this still required?

**Review by frankvHoof93 — APPROVED** [review]
Looks fine. Needs some minor cleanup

**Review by Thorvers — APPROVED** [review]

**Pull request merged**